ang/civicase-base/decorators/crm-api.decorator.js

```diff
@@ -0,0 +1,35 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase-base');
+
+  /**
+   * The original crmApi, which is part of `crmUtil` module, does not 'reject'
+   * the promise when an array of api requests are sent, and one of them got
+   * returned with error message. This decorator fixes that.
+   */
+  module.decorator('crmApi', function ($delegate, $q) {
+    var crmApi = function (entity, action, params, message) {
+      var deferred = $q.defer();
+
+      $delegate(entity, action, params, message)
+        .then(function (result) {
+          var isError = false;
+
+          if (_.isArray(result)) {
+            _.each(result, function (response) {
+              if (response.is_error) {
+                isError = true;
+              }
+            });
+          }
```

Comment:
what happens when the response is not an array but there's an error?

Reply:
That is already handled by core, https://github.com/civicrm/civicrm-core/blob/master/ang/crmUtil.js#L37